pallets/vc-management/src/lib.rs

```diff
@@ -165,5 +239,101 @@ pub mod pallet {
 			Self::deposit_event(Event::SomeError { func, error });
 			Ok(Pays::No.into())
 		}
+
+		// Change the schema Admin account
+		#[pallet::weight(195_000_000)]
+		pub fn set_schema_admin(
+			origin: OriginFor<T>,
+			new: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			T::SetAdminOrigin::ensure_origin(origin)?;
+			Self::deposit_event(Event::SchemaAdminChanged {
+				old_admin: Self::schema_admin(),
+				new_admin: Some(new.clone()),
+			});
+			<SchemaAdmin<T>>::put(new);
+			Ok(Pays::No.into())
+		}
+
+		// It requires the schema Admin account
+		#[pallet::weight(195_000_000)]
+		pub fn add_schema(
+			origin: OriginFor<T>,
+			shard: ShardIdentifier,
+			id: Vec<u8>,
+			content: Vec<u8>,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+			ensure!(Some(sender.clone()) == Self::schema_admin(), Error::<T>::RequireSchemaAdmin);
+			ensure!((id.len() as u32) <= SCHEMA_ID_LEN, Error::<T>::LengthMismatch);
+			ensure!((content.len() as u32) <= SCHEMA_CONTENT_LEN, Error::<T>::LengthMismatch);
+
+			let index = Self::schema_index();
+			let new_index = index.checked_add(1u64).ok_or(Error::<T>::SchemaIndexOverFlow);
+			if new_index.is_ok() {
+				<SchemaRegistryIndex<T>>::put(new_index.as_ref().unwrap());
+			}
+			SchemaRegistry::<T>::insert(
+				index,
+				VCSchema::<T>::new(id.clone(), sender.clone(), content.clone()),
+			);
+			Self::deposit_event(Event::SchemaIssued { account: sender, shard, index });
```

Comment:
A bit problem in this logic: if `new_index` is somehow not ok, it will still insert a new entry with key = `index`, which overwrites the existing one and emits the event. 

We need to stop it whenever an error happens, maybe use `?` here

Reply:
If in doubts we should add a test case for it

Reply:
fixed